enhancements/ingress/logging-headers.md

```diff
@@ -0,0 +1,260 @@
+```
+
+Omitting the `spec.logging.access.httpLogFormat` field to use the default format
+string would produce log messages of the following form:
+
+```
+2020-06-10T17:43:59.327608+00:00 router-default-65dccc49cc-2cshx router-default-65dccc49cc-2cshx haproxy[71]: 174.19.21.82:33634 [10/Jun/2020:17:43:59.323] fe_sni~ be_secure:openshift-console:console/pod:console-64ff858c-ghp2k:console:10.128.0.47:8443 0/0/0/3/3 303 975 - - --VN 23/5/0/0/0 0/0 {console-openshift-console.apps.ci-ln-8cjiqnb-d5d6b.origin-ci-int-aws.dev.rhcloud.com|https://console-openshift-console.apps.ci-ln-8cjiqnb-d5d6b.origin-ci-int-aws.dev.rhcloud.c} {341|https://oauth-openshift.apps.ci-ln-8cjiqnb-d5d6b.origin-ci-int-aws.dev.rhcloud.com/oauth/a} "GET /auth/login HTTP/1.1"
+```
+
+#### Validation
+
+Omitting `spec.logging.access.httpCaptureHeaders` or both
+`spec.logging.access.httpCaptureHeaders.request` and
+`spec.logging.access.httpCaptureHeaders.response` specifies the default
+behavior.
+
+The API validates the `spec.logging.access.httpCaptureHeaders.request[*].name`
+and `spec.logging.access.httpCaptureHeaders.response[*].name` field values as
+described by the field type's `+kubebuilder:validation:Pattern` marker.
+Similarly, the API validates that the
+`spec.logging.access.httpCaptureHeaders.request[*].maxLength` and
+`spec.logging.access.httpCaptureHeaders.response[*].maxLength` field values are
+integers.
+
+### User Stories
+
+#### As a cluster administrator, I need to configure an IngressController to capture certain HTTP request and response headers, in order to satisfy compliance requirements.
+
+To satisfy this use-case, the cluster administrator can set the
+IngressController's `spec.logging.access.httpCaptureHeaders` field to capture
+the required headers..
+
+### Implementation Details
+
+Implementing this enhancement requires changes in the following repositories:
+
+* openshift/api
+* openshift/cluster-ingress-operator
+* openshift/router
+
+The router configures HAProxy using a configuration template.  The template uses
+environment variables as input parameters.  The enhancement adds two new
+environment variables, `ROUTER_CAPTURE_HTTP_REQUEST_HEADERS` and
+`ROUTER_CAPTURE_HTTP_RESPONSE_HEADERS`, which specify the headers to capture.
+The ingress operator sets this variable based on the IngressController's
+`spec.logging.access.httpCaptureHeaders` field value.
+
+The values of `ROUTER_CAPTURE_HTTP_REQUEST_HEADERS` and
+`ROUTER_CAPTURE_HTTP_RESPONSE_HEADERS` are comma-delimited lists of values of
+the form `name:maxLength`.  (Colons and commas are not allow in HTTP header
```

Comment:
s/allow/allowed

> (Colons and commas are not allow in HTTP header names, so their use unambiguously delimits values in the environment variable.)
The template translates these values into appropriate `capture request header`

Should these details be mentioned in the `maxLength` field godocs? Is there an RFC or spec that should be referenced that proves this to be true?

Reply:
The standards are a little tedious to parse:  https://github.com/openshift/router/pull/139#discussion_r443796088

I'll add a reference to RFC 2616 section 4.2.